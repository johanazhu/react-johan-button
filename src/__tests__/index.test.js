import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Button } from '../index';

describe('Button', () => {
    it('render', () => {
        const wrapper = render(<Button>foo</Button>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('type', () => {
        const wrapper = render(<Button type="primary">foo</Button>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('disabled', () => {
        const wrapper = render(<Button disabled>foo</Button>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('size', () => {
        const wrapper = render(<Button size="lg">foo</Button>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('onClick', () => {
        const onClick = jest.fn();
        const wrapper = shallow(<Button onClick={onClick}>foo</Button>);
        wrapper.simulate('click');
        expect(onClick).toBeCalled();
    });
});