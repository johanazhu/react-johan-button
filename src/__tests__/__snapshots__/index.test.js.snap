// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button disabled 1`] = `
<button
  class="johan-btn johan-btn-default johan-btn-md"
  disabled=""
  type="button"
>
  foo
</button>
`;

exports[`Button render 1`] = `
<button
  class="johan-btn johan-btn-default johan-btn-md"
  type="button"
>
  foo
</button>
`;

exports[`Button size 1`] = `
<button
  class="johan-btn johan-btn-default johan-btn-lg"
  type="button"
>
  foo
</button>
`;

exports[`Button type 1`] = `
<button
  class="johan-btn johan-btn-primary johan-btn-md"
  type="button"
>
  foo
</button>
`;
